From 601fb1fb3ddeea32dbb02b140bc356e7ef4b0c7f Mon Sep 17 00:00:00 2001
From: Li Jun <b47624@freescale.com>
Date: Thu, 16 Oct 2014 22:17:56 +0800
Subject: [PATCH 0139/5242] MLK-9617-19 usb: otg: test: show warning message
 for unsupported otg test dev

commit  f26243265e39c78ef42106d0df938cc0b6c164f3 from
https://source.codeaurora.org/external/imx/linux-imx.git

According to "USB On-The-Go and Embedded Host Automated Compliance Plan Version
1.2", chapter 6.7.23, unknown Device Not Supporting HNP:'Unsupported Device';
chapter 6.7.24 unknown Device Supporting HNP: 'Device No Response'

Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit 3fb9b462769d44073078f3b29fe68e214896ea9e)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/core/otg_whitelist.h |   13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/usb/core/otg_whitelist.h b/drivers/usb/core/otg_whitelist.h
index 039aa13..2e815e6 100644
--- a/drivers/usb/core/otg_whitelist.h
+++ b/drivers/usb/core/otg_whitelist.h
@@ -85,6 +85,19 @@ static int is_targeted(struct usb_device *dev)
 	     le16_to_cpu(dev->descriptor.idProduct) == 0x0200))
 		return 1;
 
+	/* Unknown Device Not Supporting HNP */
+	if ((le16_to_cpu(dev->descriptor.idVendor) == 0x1a0a &&
+		le16_to_cpu(dev->descriptor.idProduct) == 0x0201)) {
+		dev_warn(&dev->dev, "Unsupported Device\n");
+		return 0;
+	}
+	/* Unknown Device Supporting HNP */
+	if ((le16_to_cpu(dev->descriptor.idVendor) == 0x1a0a &&
+		le16_to_cpu(dev->descriptor.idProduct) == 0x0202)) {
+		dev_warn(&dev->dev, "Device no Responding\n");
+		return 0;
+	}
+
 	/* NOTE: can't use usb_match_id() since interface caches
 	 * aren't set up yet. this is cut/paste from that code.
 	 */
-- 
1.7.9.5

